substrate/utility

```diff
@@ -784,7 +783,26 @@ namespace substrate
 	template<typename C> SUBSTRATE_NO_DISCARD(inline SUBSTRATE_CXX14_CONSTEXPR auto data(const C &c)) -> decltype(c.data()) { return c.data(); }
 	template<typename T, std::size_t N> SUBSTRATE_NO_DISCARD(inline SUBSTRATE_CXX14_CONSTEXPR T *data(T (&array)[N]) noexcept) { return array; }
 	template<typename E> SUBSTRATE_NO_DISCARD(inline SUBSTRATE_CXX14_CONSTEXPR const E *data(std::initializer_list<E> il) noexcept) { return il.begin(); }
+#endif 
```

Comment:
Floating `#endif`?

Reply:
No? that's part of the clause starting at line 778/781. The existing `#endif` is billed to the warning re-enabling pragma that gets added in this PR.

Reply:
Oh, this is caused by the diff display being weird and showing the original `#endif` below in the middle of the addition. No worries then!